File: LeetCode-Problems/Easy/Goal-Parser-Interpretation.py
# Python
class Solution(object):
    def interpret(self, command):
        """
        :type command: str
        :rtype: str
        """
        returnedString = "";
        
        for x in range(0, len(command)):
            if (command[x:x+1] == "G"):
                returnedString += "G"
            if (x < len(command)-1):
                if (command[x:x+2] == "()"):
                    returnedString += "o"
            if (x < len(command)-3):
                if (command[x:x+4] == "(al)"):
                    returnedString += "al"
        return returnedString

File: LeetCode-Problems/Easy/Goal-Parser-Interpretation.java
//Java
class Solution {
    public String interpret(String command) {
        String returnedString = "";
        for (int i = 0; i < command.length(); i++){
            if (command.substring(i, i+1).equals("G")){
                returnedString += "G";
            }
            if (i < command.length()-1){
                if (command.substring(i, i+2).equals("()")){
                    returnedString += "o";
                }
            }
            if (i < command.length()-3){
                if (command.substring(i, i+4).equals("(al)")){
                    returnedString += "al";
                }
            }
        }
        return returnedString;
    }
}
